### PR TITLE
Added ByteString benchmarks

### LINKED ISSUE #153: Extend benchmarking suite with ByteString-related benchmarks

### PR DESCRIPTION
**Pull request by fzhinkin**
Added benchmarks covering the following functions:
- indexOf
- lastIndexOf
- startsWith
- endsWith
- hashCode
- compareTo

Fixes #153

### CODE REVIEW

File: benchmarks/src/commonMain/kotlin/BufferOps.kt [hunk 1]
```diff
@@ -384,3 +385,33 @@ open class BufferReadNewByteArray : BufferRWBenchmarkBase() {
...
+            if (idx % bsSize == 0) {
+                buffer.writeByte(0)
+            } else {
+                buffer.writeByte(0x42)
+            }
+        }
+    }
+
+    @Benchmark
+    fun benchmark() = buffer.indexOf(byteString)
```
Comment (shanshin) [inline benchmarks/src/commonMain/kotlin/BufferOps.kt:416]: In this case, the result of `indexOf` will always be 0?

File: benchmarks/src/commonMain/kotlin/bytestring/ByteStringOps.kt [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    var params: String = "<string length>:<prefix/suffix length>:<mismatch offset, or -1>"
+
+    override fun getRawParams(): String = params
+
+    @Benchmark
+    fun benchmark() = byteString.endsWith(targetByteString)
+}
+
+@State(Scope.Benchmark)
+open class CompareBenchmark {
```
Comment (shanshin) [inline benchmarks/src/commonMain/kotlin/bytestring/ByteStringOps.kt:191]: Should we also add tests on `equals`?

File: benchmarks/src/commonMain/kotlin/BufferOps.kt (hunk 1 above)
Comment (fzhinkin) [reply benchmarks/src/commonMain/kotlin/BufferOps.kt:416]: In fact, every byte at a position whose index is a multiple of bytestring size is `0`, so the benchmark has to scan the whole buffer, every time looking for up to `length-1` bytes ahead.

File: benchmarks/src/commonMain/kotlin/bytestring/ByteStringOps.kt (hunk 2 above)
Comment (fzhinkin) [reply benchmarks/src/commonMain/kotlin/bytestring/ByteStringOps.kt]: Yes, indeed. Added the benchmark.

**Review by shanshin — APPROVED** [review]

**Pull request merged**